File: SDKs/apideck.com/ecosystem/0.0.6/python/src/sdk/models/operations/productsall.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared


@dataclass
class ProductsAllPathParams:
    ecosystem_id: str = field(metadata={'path_param': { 'field_name': 'ecosystem_id', 'style': 'simple', 'explode': False }})
    

@dataclass
class ProductsAllRequest:
    path_params: ProductsAllPathParams = field()
    

@dataclass
class ProductsAllResponse:
    content_type: str = field()
    status_code: int = field()
    get_products_response: Optional[shared.GetProductsResponse] = field(default=None)
    

File: SDKs/apideck.com/ecosystem/0.0.6/typescript/src/sdk/models/operations/productsall.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class ProductsAllPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=ecosystem_id" })
  ecosystemId: string;
}


export class ProductsAllRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: ProductsAllPathParams;
}


export class ProductsAllResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  getProductsResponse?: shared.GetProductsResponse;

  @SpeakeasyMetadata()
  statusCode: number;
}
